lsp/src/main/java/org/aya/lsp/utils/Resolver.java

```diff
@@ -156,7 +157,9 @@
     }
 
     @Override public void visitSignatured(@NotNull Signatured signatured, XY xy) {
-      signatured.telescope.forEach(tele -> check(xy, tele.ref(), tele.sourcePos()));
+      signatured.telescope
+        .filterNot(tele -> tele.ref().name().startsWith(Constants.ANONYMOUS_PREFIX))
```

Comment:
What if this name is written by users?

Reply:
We should reserve the meaning of names starting with an underscore. It means _does not matter_. If users any term uses those names the resolver should raise an warning or error.

Reply:
Is this just a performance improvement? What if there's a lambda expression in an anonymous parameter?

Reply:
> Is this just a performance improvement?

Yes, finding usages on each LSP action (like hover) is very slow.

> What if there's a lambda expression in an anonymous parameter?

I don't get it. Care to give an example?

Reply:
> We should reserve the meaning of names starting with an underscore. It means _does not matter_. If any term uses those names the resolver should raise a warning or even an error.

I agree

Reply:
> I don't get it. Care to give an example?

Like `def test (Ok (\ x => x))`, the above code implies that this `Ok (\ x => x)` will not be searched in this visitor. Is this intended?

Reply:
> > I don't get it. Care to give an example?
> 
> Like `def test (Ok (\ x => x))`, the above code implies that this `Ok (\ x => x)` will not be searched in this visitor. Is this intended?

No, the type is always searched

Reply:
This PR fixes searching `PsiNamedElement` alike things, that is an anonymous parameter never introduces a usable name.

Reply:
Oh I see, it's searched in `super.visitSignatured`, tyvm!

Reply:
bors r+